src/App.tsx

```diff
@@ -8,39 +8,37 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { SelectedPost } from './features/selectedpost/selectedPost';
+import { Author } from './features/author/authorSlice';
+import * as selectedPostActions from './features/selectedpost/selectedPost';
+import * as authorActions from './features/author/authorSlice';
+import * as postsActions from './features/posts/postsSlice';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
-
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
-
-  function loadUserPosts(userId: number) {
-    setLoaded(false);
-
-    getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
-  }
+  const { posts, loaded, hasError } = useAppSelector(state => state.posts);
+  const author = useAppSelector(state => state.author);
+  const selectedPost = useAppSelector(state => state.selectedPost);
```

Comment:
All above declarations could be merged into a single one:
```
const {
    posts: {
        posts,
        loaded,
        hasError
    },
    author,
    selectedPost
} = useAppSelector(state => state);  //maybe even state => state can be omitted
```

Reply:
tks @pedro-ruas , i'll keep that in mind for next projects.